src/txrebroadcast.cpp

```diff
@@ -181,3 +181,29 @@ void TxRebroadcastHandler::TrimMaxRebroadcast()
         m_attempt_tracker->get<index_by_last_attempt>().erase(it);
     }
 };
+
+void TxRebroadcastHandler::UpdateAttempt(const uint256 txhsh, const int count, const std::chrono::microseconds last_attempt_time)
+{
+    auto it = m_attempt_tracker->find(txhsh);
+    for (int i = 0; i < count; ++i) {
+        auto UpdateRebroadcastEntry = [last_attempt_time](RebroadcastEntry& rebroadcast_entry) {
+            rebroadcast_entry.m_last_attempt = last_attempt_time;
+            ++rebroadcast_entry.m_count;
+        };
+
+        m_attempt_tracker->modify(it, UpdateRebroadcastEntry);
+    }
```

Comment:
In d579110263ad07386f1fbf796a3397a12340b6d5 "[test] Add unit test for rebroadcast attempt logic"

It seems like this for loop is not necessary as it is essentially doing `m_count += count`. Could instead be:

```suggestion
    auto UpdateRebroadcastEntry = [last_attempt_time, count](RebroadcastEntry& rebroadcast_entry) {
        rebroadcast_entry.m_last_attempt = last_attempt_time;
        rebroadcast_entry.m_count += count;
    };

    m_attempt_tracker->modify(it, UpdateRebroadcastEntry);
```

Reply:
good point, updated!